src/nc_core.c

```diff
@@ -261,6 +292,41 @@ core_error(struct context *ctx, struct conn *conn)
     core_close(ctx, conn);
 }
 
+static void
+retry_connection(struct context *ctx)
+{
+    struct array *servers;
+    int idx;
+    struct server *server;
+    int64_t now;
+    uint32_t i, nsize;
+    rstatus_t status;
+
+    servers = ctx->fails;
+    idx = (ctx->failed_idx == 0) ? 1 : 0;
+
+    ctx->failed_idx = idx;
+    ctx->fails = &(ctx->failed_servers[idx]);
+
+    now = nc_usec_now();
```

Comment:
nit: can move this below the `if (nsize == 0)` to avoid the call to gettimeofday when no servers fail